Library/Homebrew/formula_installer.rb

```diff
@@ -114,7 +114,7 @@ def pour_bottle?(install_bottle_options = { warn: false })
 
   def install_bottle_for?(dep, build)
     return pour_bottle? if dep == formula
-    return false if build_from_source?
+    return false if ARGV.build_formula_from_source?(dep)
```

Comment:
I don't know. The existing `build_from_source?` method is an accessor on `@build_from_source`, which is twiddled from `cmd/install.rb` and the `DependencyInstaller` subclass. So unifying them would be a more extensive refactor of `FormulaInstaller` and `cmd/install.rb`. Could simplify things, but I'm reluctant to do the refactoring here, because it would change the control flow pattern for `Homebrew#install` a bit. The context in which they're used (the main formula vs a discovered dependency) is different enough that the existing control flow seems more readable to me.